Dependencies/xr/Source/OpenXR/SceneUnderstanding.cpp

```diff
@@ -375,6 +441,17 @@ struct SceneUnderstanding::SceneUnderstanding::Impl
                 babylonPlane.Polygon[11] = 0;
                 babylonPlane.PolygonSize = babylonPlane.Polygon.size() / VALUES_IN_POINT; // Number of points
 
+                babylonPlane.ParentGeometryId = static_cast<xr::GeometryId>(objectKey);
+
+                if (s_sceneObjectTypeMap.count(object->Kind) > 0)
+                {
+                    babylonPlane.ParentGeometryType = s_sceneObjectTypeMap.at(object->Kind);
+                }
+                else
+                {
+                    babylonPlane.ParentGeometryType = s_sceneObjectTypeMap.at(XrSceneObjectKindTypeMSFT::XR_SCENE_OBJECT_KIND_TYPE_UNKNOWN_MSFT);

```

Comment:
nit: I think this file might be exclusively intended for exploring capabilities which are currently Microsoft-specific, in which case this is already in an area where it won't clash. But just in case, if I'm incorrect and this is in a code path that other vendors/platforms may encounter, we should be proactive about preparing to have other use cases coexist alongside our vendor-specific implementations.

Reply:
scene understanding is currently a microsoft specific preview extension. I think there's hope that long term it will become a khronos group extension that other vendors implement. I would guess that at this time, the XrSceneObjectKindTypeMSFT will be renamed.